src/AutoMapper/MappingEngine.cs

```diff
@@ -410,40 +382,34 @@ protected override Expression VisitMember(MemberExpression node)
         }
 
 
-        object IMappingEngineRunner.Map(ResolutionContext context)
+        public object Map(ResolutionContext context)
         {
             try
             {
+                if (context.TypeMap != null)
+                {
+                    context.TypeMap.Seal();

```

Comment:
This is not thread safe see #1010 for example. And anyway, Seal should have been called on init, no?